### PR TITLE
Inherit `@immutable` phpdoc

### PR DESCRIPTION
**Pull request by herndlm**
Closes https://github.com/phpstan/phpstan/issues/7506

### CODE REVIEW

**Comment by BackEndTea** [comment]
Should this also work for interfaces? E.g. declare @immutable on the interface means all implementations are immutable

**Comment by herndlm** [comment]
> Should this also work for interfaces? E.g. declare @immutable on the interface means all implementations are immutable

I might be missing something, but I think it should be fine because immutable currently only makes all props readonly and interfaces don't have props, right?
Methods are by default considered to be pure AFAIK and therefore they are not considered via immutable yet.

File: src/Reflection/ClassReflection.php [hunk 1]
```diff
@@ -1047,6 +1047,11 @@ public function isImmutable(): bool
 			$this->isImmutable = $resolvedPhpDoc !== null && $resolvedPhpDoc->isImmutable();
 		}
 
+		$parentClass = $this->getParentClass();
+		if ($parentClass !== null && !$this->isImmutable) {
```
Comment (ondrejmirtes) [inline src/Reflection/ClassReflection.php]: Please move everything into the `if ($this->isImmutable === null) {` code block - the whole logic of this method is that the property caches the result. The computation should be done inside that code block.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**